syntax = "proto3";

package helloworld;

service Greeter {
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc SayHelloClientStreaming (stream HelloRequest) returns (HelloReply) {}
    rpc SayHelloServerStreaming (HelloRequest) returns (stream HelloReply) {}
    rpc SayHelloBidirectionalStreaming (stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}
